core/src/main/java/com/yahoo/oak/EntryHashSet.java

```diff
@@ -665,7 +675,14 @@ boolean isEntryDeleted(ValueBuffer tempValue, int ei) {
         return tempValue.getSlice().isDeleted() != ValueUtils.ValueResult.FALSE;
     }
 
-
+    @Override
+    int getNextNonZeroIndex(int currentIndex) {
+        int nxtIdx = mapOfCleanEntries.nextSetBit(currentIndex + 1);
```

Comment:
Why do you think it is OK not to use synchronization while reading?

Reply:
if several processes read read the value at the same time, could it be a problem?

Reply:
Generally, shouldn't be, but it depends how `BitSet` is implemented. If the update is more than a machine word, such an unprotected read may read partially updated structure. Do we know at least approximately whether BitSet is array with relevant bits set or may be a linked list with nodes, for example?

Reply:
No, we don't know how it is implemented, we don't want to guess either. So, the synchronisation on function level is not enough, we need to add a lock and acquire it before every write/read on the bitset